src/realm/sync/client_base.hpp

```diff
@@ -285,12 +285,27 @@ inline std::ostream& operator<<(std::ostream& os, ConnectionState state)
 
 // The reason a synchronization session is used for.
 enum class SessionReason {
-    // Regular synchronization
+    // Regular synchronization => BIND message sessionReason = 0
     Sync = 0,
-    // Download a fresh realm
-    ClientReset,
+    // Download a fresh realm => BIND message sessionReason = 1
```

Comment:
This seems to be changing this type to conflate two completely unrelated bits of state. This enum is naming the values passed to the server for the session reason, and adding other states doesn't make much sense.

Reply:
Updated - now this enum is back to the original 2 values and a separate bool member variable (`m_fresh_realm_download`) was added to `ClientImpl::Session` that is set to true for client reset fresh download sessions in `Activate()`

The original SessionReason couldn't be used as-is since it is set to `ClientReset` for both the fresh download and client reset diff sessions...